### PR TITLE
chore: use `tweag/run-nix-shell`

### PR DESCRIPTION
**Pull request by cgrindel**
Related to https://github.com/tweag/scalable-builds-group/issues/138.

### CODE REVIEW

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -96,29 +98,35 @@ jobs:
...
       - name: Build & test - rules_haskell_nix
         if: matrix.module == 'rules_haskell_nix'
-        run: |
-          nix-shell --arg docTools false --argstr ghcVersion ${{ matrix.ghc }} --pure --run '
+        uses: tweag/run-nix-shell@v0
+        with:
+          options: ${{ env.NIX_SHELL_ARGS }}
+          run: |
             set -euo pipefail
             cd rules_haskell_nix
```
Comment (avdv) [inline .github/workflows/workflow.yaml:122]: Could we use `working-directory:` instead and drop the `cd command` and `set -euo pipefail`?

File: .github/workflows/workflow.yaml [hunk 2]
```diff
@@ -96,29 +98,35 @@ jobs:
...
       - name: Build & test - rules_haskell_tests
         if: matrix.module == 'rules_haskell_tests'
-        run: |
-          nix-shell --arg docTools false --argstr ghcVersion ${{ matrix.ghc }} --pure --run '
+        uses: tweag/run-nix-shell@v0
+        with:
+          options: ${{ env.NIX_SHELL_ARGS }}
+          run: |
             set -euo pipefail
             cd rules_haskell_tests
```
Comment (avdv) [inline .github/workflows/workflow.yaml:131]: Likewise, but keep the `set -euo pipefail` since this runs multiple commands.

**Review by avdv — CHANGES_REQUESTED** [review]
Thank you, just some minor remarks.

**Review by avdv — APPROVED** [review]
Thank you!

File: .github/workflows/workflow.yaml (hunk 1 above)
Comment (cgrindel) [reply .github/workflows/workflow.yaml]: It looks like I need to add an attribute to `run-nix-shell` to specify the location of `shell.nix`. I just added [an issue](https://github.com/tweag/run-nix-shell/issues/4) to track. Are you OK with me leaving the `cd ...` for now?

**Pull request merged**